aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -366,36 +362,48 @@ public class AwsProperties implements Serializable {
    */
   public static final String HTTP_CLIENT_TYPE_URLCONNECTION = "urlconnection";
 
+  /**
+   * The prefix for configurations of for {@link
+   * software.amazon.awssdk.http.urlconnection.UrlConnectionHttpClient.Builder}
+   */
+  public static final String HTTP_CLIENT_URLCONNECTION_PREFIX = "http-client.urlconnection.";
+
   /**
    * If this is set under {@link #HTTP_CLIENT_TYPE}, {@link
    * software.amazon.awssdk.http.apache.ApacheHttpClient} will be used as the HTTP Client in {@link
    * AwsClientFactory}
    */
   public static final String HTTP_CLIENT_TYPE_APACHE = "apache";
 
+  /**
+   * The prefix for configurations of for {@link
+   * software.amazon.awssdk.http.apache.ApacheHttpClient.Builder}
+   */
+  public static final String HTTP_CLIENT_APACHE_PREFIX = "http-client.apache.";
+
   public static final String HTTP_CLIENT_TYPE_DEFAULT = HTTP_CLIENT_TYPE_URLCONNECTION;
 
   /**
    * Used to configure the connection timeout in milliseconds for {@link
-   * UrlConnectionHttpClient.Builder}. This flag only works when {@link #HTTP_CLIENT_TYPE} is set to
-   * {@link #HTTP_CLIENT_TYPE_URLCONNECTION}
+   * software.amazon.awssdk.http.urlconnection.UrlConnectionHttpClient.Builder}. This flag only
+   * works when {@link #HTTP_CLIENT_TYPE} is set to {@link #HTTP_CLIENT_TYPE_URLCONNECTION}
    *
    * <p>For more details, see
    * https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/http/urlconnection/UrlConnectionHttpClient.Builder.html
    */
   public static final String HTTP_CLIENT_URLCONNECTION_CONNECTION_TIMEOUT_MS =
-      "http-client.urlconnection.connection-timeout-ms";
+      HTTP_CLIENT_URLCONNECTION_PREFIX + HttpClientConfigurations.CONNECTION_TIMEOUT_MS;
```

Comment:
I remember we preferred to use the full config name because it is easier for people to search for it in the codebase

Reply:
So what we can do is that, we can keep these strings as is, and then instead of using `propertiesWithPrefix`, we can just pass in the original property map to initialize the client

Reply:
totally agree with @jackye1995 here.

Reply:
Thank you for your suggestions. I've restored them to full config names